File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Flora.java
package com.logistics.Components.Product;

public class Flora extends Product {

    protected boolean isCultivated;

    public Flora() {
        super();
        isCultivated = false;
    }

    public Flora(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, boolean isCultivated) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit);
        this.isCultivated = isCultivated;
    }

    // Copy Constructor
    public Flora(Flora other) {
        super(other);
        this.isCultivated = other.isCultivated;
    }

    // Clone method
    @Override
    public Flora clone() {
        return new Flora(this);
    }

    // Getters
    public boolean getIsCultivated() { return isCultivated; }

    // Setters
    public void setIsCultivated(boolean isCultivated) { this.isCultivated = isCultivated; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nCultivated: " + isCultivated;
    }
}

File: Task 3/C++Stock Viewer/Product/Flora.cpp
#include "Flora.h"

Flora::Flora() : Product(){
    isCultivated = false;
}

Flora::Flora(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, bool isCultivated) 
        : Product(name, cost, supplier, discount_amount, packing_units, cost_per_unit){
    this->isCultivated = isCultivated;
}

Flora::Flora(const Flora& other) : Product(other) {
    this->isCultivated = other.isCultivated;
}

std::string Flora::display_product_info(){
    return Product::display_product_info() +
            "\nIs Cultivated: "+(isCultivated? "true" : "false");
}